### PR TITLE
ARXIVNG-1301 enforce new submission size limits

### PR DESCRIPTION
**Pull request by erickpeirson**
This implements the following size limits via QA rules hooked to ``Set/UpdateSourceContent`` events.

- Compressed package: 6MB
- Uncompressed package: 18MB
- Individual PDF: 15MB

I cleaned up a bunch of testing cruft and made other little tweaks, so the number of files changed looks artificially high. See comments below for pointers to the real meat of the PR.

### CODE REVIEW

File: core/arxiv/submission/rules/size_limits.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+"""Enforces size limit rules."""
```
Comment (erickpeirson) [inline core/arxiv/submission/rules/size_limits.py]: This is the meat of the PR

File: core/arxiv/submission/tests/examples/test_09_compilation.py [hunk 2]
```diff
@@ -0,0 +1,362 @@
+"""Compilation workflow during submission."""
```
Comment (erickpeirson) [inline core/arxiv/submission/tests/examples/test_09_compilation.py]: These are the main tests that were added

File: core/arxiv/submission/rules/size_limits.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+@UpdateUploadPackage.bind(lambda *args, **kwargs: True)    # Always check.
+def check_sizes_on_update_source(event: UpdateUploadPackage,
+                                 before: Submission, after: Submission,
+                                 creator: Agent) -> Iterable[Event]:
+    """When a source package is updated, check for oversize source."""
+    return _check_sizes(event, before, after, creator)
+
+
+def pdf_is_compiled(event: Event, *args, **kwargs) -> bool:
```
Comment (JaimieMurdock) [inline core/arxiv/submission/rules/size_limits.py]: Should this method be bound to a particular event?

Comment (erickpeirson) [reply core/arxiv/submission/rules/size_limits.py]: This is use as the condition for the callback, on L51. This way we are only running ``check_pfdf_size`` when the process being updated is for compilation, and the status is ``SUCCEEDED``.

File: core/arxiv/submission/tests/examples/test_09_compilation.py (hunk 2 above)
Comment (DavidLFielding) [reply core/arxiv/submission/tests/examples/test_09_compilation.py]: Tests run fine for me and code appear to exercise compressed and uncompressed limits.

**Comment by erickpeirson** [comment]
@DavidLFielding Thanks for the detailed thoughts on the size limits. These are quite valid points, and I'll follow up separately about how we can move that conversation forward.

Regarding where the compressed size will comes from: I assumed we'd just use the size of the tarball from the FM service, as you mention. Will add a note to implement that before closing ARXIVNG-1301.

**Pull request merged**